### PR TITLE
[5.3] Template Override Plugins Compare

### PR DESCRIPTION
**Pull request by brianteeman**
### Summary of Changes
Adds override compare functionality to plugin overrides

### Testing Instructions
In the template manager create an override for any plugin
Select one of the files in the override

### Actual result BEFORE applying this Pull Request
Show Original File and Show Differences not present

image

### Expected result AFTER applying this Pull Request
Show Original File and Show Differences are present and you can se the differences etc

image

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [x] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on ab53b22fdb069e1f9707a6421d044b07232b184b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44588">issues.joomla.org/tracker/joomla-cms/44588</a>.</sub>

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 1]
```diff
@@ -500,6 +501,18 @@ public function getCoreFile($file, $client_id)
...
+            $subFolder       = $pluginFolder['2'];
+            $htmlPath        = Path::clean($pluginPath . $folder . '/' . $subFolder . '/tmpl/');
+            $fileName        = $this->getSafeName($fileName);
+            $coreFile        = Path::find($htmlPath, $fileName);
+
+            return $coreFile;
+        }
+
         if (stristr($type, 'com_') !== false) {
             // For components
```
Comment (sandewt) [inline administrator/components/com_templates/src/Model/TemplateModel.php:517]:
```suggestion
        // For components
        if (stristr($type, 'com_') !== false) {
```

Comment (sandewt) [reply administrator/components/com_templates/src/Model/TemplateModel.php:517]: See also line 537: ```// For Layouts```

File: administrator/components/com_templates/src/Model/TemplateModel.php [hunk 2]
```diff
@@ -500,6 +501,18 @@ public function getCoreFile($file, $client_id)
...
         }
 
+        // For plugins
+        if (stristr($type, 'plg_') !== false) {
+            $pluginFolder    = explode('_', $explodeArray['2']);
+            $folder          = $pluginFolder['1'];
+            $subFolder       = $pluginFolder['2'];
+            $htmlPath        = Path::clean($pluginPath . $folder . '/' . $subFolder . '/tmpl/');
+            $fileName        = $this->getSafeName($fileName);
+            $coreFile        = Path::find($htmlPath, $fileName);
```
Comment (sandewt) [inline administrator/components/com_templates/src/Model/TemplateModel.php]: Indent spaces
```suggestion
            $pluginFolder = explode('_', $explodeArray['2']);
            $folder       = $pluginFolder['1'];
            $subFolder    = $pluginFolder['2'];
            $htmlPath     = Path::clean($pluginPath . $folder . '/' . $subFolder . '/tmpl/');
            $fileName     = $this->getSafeName($fileName);
            $coreFile     = Path::find($htmlPath, $fileName);
```

File: administrator/components/com_templates/src/Model/TemplateModel.php (hunk 1 above)
Comment (brianteeman) [reply administrator/components/com_templates/src/Model/TemplateModel.php:517]: beyond the scope of this pr

**Comment by richard67** [comment]
I've restored @fgsw 's test result in the issue tracker and have set the test result for @Quy as he can't do that currently so asked in Mattermost to do that.

=> RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/44588">issues.joomla.org/tracker/joomla-cms/44588</a>.</sub>

**Comment by HLeithner** [comment]
slightly related is the wrong type usage for array access `$explodeArray['1']` casts in implizit type conversation to integer which is not correct here. So if someone likes to fix this please go ahead.

**Pull request merged**